Reland: [clang] Improved canonicalization for template specialization types

mizvekov opened the pull request:
This relands https://github.com/llvm/llvm-project/pull/135119, after fixing crashes seen in LLDB CI reported here: https://github.com/llvm/llvm-project/pull/135119#issuecomment-2794910840

Fixes https://github.com/llvm/llvm-project/pull/135119

This changes the TemplateArgument representation to hold a flag indicating whether a tempalte argument of expression type is supposed to be canonical or not.

This gets one step closer to solving https://github.com/llvm/llvm-project/issues/92292

This still doesn't try to unique as-written TSTs. While this would increase the amount of memory savings and make code dealing with the AST more well-behaved, profiling template argument lists is still too expensive for this to be worthwhile, at least for now.

This also fixes the context creation of TSTs, so that they don't in some cases get incorrectly flagged as sugar over their own canonical form. This is captured in the test expectation change of some AST dumps.

This fixes some places which were unnecessarily canonicalizing these TSTs.

mizvekov committed:
Reland: [clang] Improved canonicalization for template specialization types

This changes the TemplateArgument representation to hold a flag indicating
whether a tempalte argument of expression type is supposed to be canonical
or not.

This gets one step closer to solving https://github.com/llvm/llvm-project/issues/92292

This still doesn't try to unique as-written TSTs. While this would
increase the amount of memory savings and make code dealing with
the AST more well-behaved, profiling template argument lists is
still too expensive for this to be worthwhile, at least for now.

This also fixes the context creation of TSTs, so that they don't
in some cases get incorrectly flagged as sugar over their own canonical
form. This is captured in the test expectation change of some AST dumps.

This fixes some places which were unnecessarily canonicalizing these TSTs.

mizvekov commented:
Adding lldb back to the ci script, as otherwise lldb cannot be tested locally.

slydiman commented:
Note [lldb-x86_64-debian](https://lab.llvm.org/buildbot/#/builders/162) and [lldb-aarch64-windows](https://lab.llvm.org/buildbot/#/builders/141) did not fail because used gcc and msvc to build lldb. But failed lldb tests required clang. I will test this patch on the environment similar to [lldb-remote-linux-win](https://lab.llvm.org/buildbot/#/builders/197).

slydiman commented:
I have tested the initial commit https://github.com/llvm/llvm-project/pull/135414/commits/f8e6b60276eabe0d89db03df95af1315590b78b9

The issue is still there
```
Assertion failed: Arg.structurallyEquals(getCanonicalTemplateArgument(Arg)), file D:\as\mainline\llvm-project\clang\lib\AST\ASTContext.cpp, line 5619
 Windows fatal exception: code 0x80000003

Current thread 0x00006b58 (most recent call first):
  File "D:\test\build-lldb\Lib\site-packages\lldb\__init__.py", line 6539 in EvaluateExpression
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 2547 in expect_expr
  File "D:\test\llvm-project\lldb\test\API\commands\expression\import-std-module\array\TestArrayFromStdModule.py", line 29 in test
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\decorators.py", line 148 in wrapper
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1804 in test_method
  File "C:\Python312\Lib\unittest\case.py", line 589 in _callTestMethod
  File "C:\Python312\Lib\unittest\case.py", line 634 in run
  File "C:\Python312\Lib\unittest\case.py", line 690 in __call__
  File "C:\Python312\Lib\unittest\suite.py", line 122 in run
  File "C:\Python312\Lib\unittest\suite.py", line 84 in __call__
  File "C:\Python312\Lib\unittest\suite.py", line 122 in run
  File "C:\Python312\Lib\unittest\suite.py", line 84 in __call__
  File "C:\Python312\Lib\unittest\runner.py", line 240 in run
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\dotest.py", line 1108 in run_suite
  File "D:\test\llvm-project\lldb\test\API\dotest.py", line 8 in <module>
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Exception Code: 0x80000003
  #0 0x00007ffd386632f5 PyInit__lldb (D:\test\build-lldb\Lib\site-packages\lldb\_lldb.cp312-win_amd64.pyd+0x9f32f5)
  #1 0x00007ffe13a71989 (C:\WINDOWS\System32\ucrtbase.dll+0xc1989)
  #2 0x00007ffe13a54ab1 (C:\WINDOWS\System32\ucrtbase.dll+0xa4ab1)
  #3 0x00007ffe13a72986 (C:\WINDOWS\System32\ucrtbase.dll+0xc2986)
  #4 0x00007ffe13a72b61 (C:\WINDOWS\System32\ucrtbase.dll+0xc2b61)
  #5 0x00007ffd3af1f959 PyInit__lldb (D:\test\build-lldb\Lib\site-packages\lldb\_lldb.cp312-win_amd64.pyd+0x32af959)
  #6 0x00007ffd3b18d481 PyInit__lldb (D:\test\build-lldb\Lib\site-packages\lldb\_lldb.cp312-win_amd64.pyd+0x351d481)
...
```
I will rebuild the debug version to get the useful stack trace.

mizvekov commented:
Thanks, yeah this relanding was just to try to get LLDB CI up and running, which I am still trying.

slydiman commented:
I tried the single test
`C:/Python312/python_d.exe D:/test/llvm-project/lldb/test/API/dotest.py ... D:/test/llvm-project/lldb/test/API/commands/expression/import-std-module/array -p TestArrayFromStdModule.py`

Here is the debug stack trace:
```
Assertion failed: Arg.structurallyEquals(getCanonicalTemplateArgument(Arg)), file D:\test\llvm-project\clang\lib\AST\ASTContext.cpp, line 5619
 Windows fatal exception: code 0x80000003
 
Current thread 0x00006a18 (most recent call first):
  File "D:\test\build-lldb\Lib\site-packages\lldb\__init__.py", line 6539 in EvaluateExpression
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 2547 in expect_expr
  File "D:\test\llvm-project\lldb\test\API\commands\expression\import-std-module\array\TestArrayFromStdModule.py", line 29 in test
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\decorators.py", line 148 in wrapper
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\lldbtest.py", line 1804 in test_method
  File "C:\Python312\Lib\unittest\case.py", line 589 in _callTestMethod
  File "C:\Python312\Lib\unittest\case.py", line 634 in run
  File "C:\Python312\Lib\unittest\case.py", line 690 in __call__
  File "C:\Python312\Lib\unittest\suite.py", line 122 in run
  File "C:\Python312\Lib\unittest\suite.py", line 84 in __call__
  File "C:\Python312\Lib\unittest\suite.py", line 122 in run
  File "C:\Python312\Lib\unittest\suite.py", line 84 in __call__
  File "C:\Python312\Lib\unittest\runner.py", line 240 in run
  File "D:\test\llvm-project\lldb\packages\Python\lldbsuite\test\dotest.py", line 1108 in run_suite
  File "D:\test\llvm-project\lldb\test\API\dotest.py", line 8 in <module>
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Exception Code: 0x80000003
  #0 0x00007ffd2b9f5bac HandleAbort D:\test\llvm-project\llvm\lib\Support\Windows\Signals.inc:429:0
  #1 0x00007ffdcdf390ed (C:\WINDOWS\SYSTEM32\ucrtbased.dll+0xa90ed)
  #2 0x00007ffdcdf3ae49 (C:\WINDOWS\SYSTEM32\ucrtbased.dll+0xaae49)
  #3 0x00007ffdcdf40c6f (C:\WINDOWS\SYSTEM32\ucrtbased.dll+0xb0c6f)
  #4 0x00007ffdcdf3eba1 (C:\WINDOWS\SYSTEM32\ucrtbased.dll+0xaeba1)
  #5 0x00007ffdcdf418af (C:\WINDOWS\SYSTEM32\ucrtbased.dll+0xb18af)
  #6 0x00007ffd33b0afe1 clang::ASTContext::getCanonicalTemplateSpecializationType(class clang::TemplateName, class llvm::ArrayRef<class clang::TemplateArgument>) const D:\test\llvm-project\clang\lib\AST\ASTContext.cpp:5619:0
  #7 0x00007ffd340cff87 clang::ASTNodeImporter::VisitTemplateSpecializationType(class clang::TemplateSpecializationType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1665:0
  #8 0x00007ffd34184739 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:77:0
  #9 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #10 0x00007ffd340b9c23 clang::ASTImporter::Import(class clang::QualType) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9180:0
 #11 0x00007ffd34156589 clang::ASTNodeImporter::import<class clang::QualType>(class clang::QualType const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #12 0x00007ffd340cffe8 clang::ASTNodeImporter::VisitTemplateSpecializationType(class clang::TemplateSpecializationType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1669:0
 #13 0x00007ffd34184739 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:77:0
 #14 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #15 0x00007ffd340bd0b3 clang::ASTImporter::Import(class clang::NestedNameSpecifier *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9795:0
 #16 0x00007ffd34161596 clang::ASTImporter::importInto<class clang::NestedNameSpecifier *>(class clang::NestedNameSpecifier *&, class clang::NestedNameSpecifier *const &) D:\test\llvm-project\clang\include\clang\AST\ASTImporter.h:310:0
 #17 0x00007ffd340bd375 clang::ASTImporter::Import(class clang::NestedNameSpecifierLoc) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9823:0
 #18 0x00007ffd3415557b clang::ASTNodeImporter::import<class clang::NestedNameSpecifierLoc>(class clang::NestedNameSpecifierLoc const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #19 0x00007ffd34160edd clang::ASTNodeImporter::importChecked<class clang::NestedNameSpecifierLoc>(class llvm::Error &, class clang::NestedNameSpecifierLoc const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
 #20 0x00007ffd34108489 clang::ASTNodeImporter::VisitDependentScopeDeclRefExpr(class clang::DependentScopeDeclRefExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8561:0
 #21 0x00007ffd3417d423 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:462:0
 #22 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #23 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
 #24 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #25 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
 #26 0x00007ffd34100c08 clang::ASTNodeImporter::VisitBinaryOperator(class clang::BinaryOperator *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7819:0
 #27 0x00007ffd34185229 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::VisitBinLAnd(class clang::BinaryOperator *) D:\test\llvm-project\clang\include\clang\AST\StmtVisitor.h:138:0
 #28 0x00007ffd3417ada6 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\llvm-project\clang\include\clang\AST\StmtVisitor.h:72:0
 #29 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #30 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
 #31 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #32 0x00007ffd34102942 clang::ASTNodeImporter::VisitImplicitCastExpr(class clang::ImplicitCastExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7973:0
 #33 0x00007ffd3417d7d9 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:534:0
 #34 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #35 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
 #36 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #37 0x00007ffd34111476 clang::ASTNodeImporter::import<class clang::TemplateArgument>(class clang::TemplateArgument const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:895:0
 #38 0x00007ffd340d7271 clang::ASTNodeImporter::ImportTemplateArguments(class llvm::ArrayRef<class clang::TemplateArgument>, class llvm::SmallVectorImpl<class clang::TemplateArgument> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:2447:0
 #39 0x00007ffd340cfe37 clang::ASTNodeImporter::VisitTemplateSpecializationType(class clang::TemplateSpecializationType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1660:0
 #40 0x00007ffd34184739 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:77:0
 #41 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #42 0x00007ffd340b9c23 clang::ASTImporter::Import(class clang::QualType) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9180:0
 #43 0x00007ffd34156589 clang::ASTNodeImporter::import<class clang::QualType>(class clang::QualType const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #44 0x00007ffd340cc6e5 clang::ASTNodeImporter::VisitElaboratedType(class clang::ElaboratedType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1682:0
 #45 0x00007ffd34184469 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:49:0
 #46 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #47 0x00007ffd340b9c23 clang::ASTImporter::Import(class clang::QualType) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9180:0
 #48 0x00007ffd34156589 clang::ASTNodeImporter::import<class clang::QualType>(class clang::QualType const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #49 0x00007ffd340cffe8 clang::ASTNodeImporter::VisitTemplateSpecializationType(class clang::TemplateSpecializationType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1669:0
 #50 0x00007ffd34184739 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:77:0
 #51 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #52 0x00007ffd340b9c23 clang::ASTImporter::Import(class clang::QualType) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9180:0
 #53 0x00007ffd34156589 clang::ASTNodeImporter::import<class clang::QualType>(class clang::QualType const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #54 0x00007ffd340cc6e5 clang::ASTNodeImporter::VisitElaboratedType(class clang::ElaboratedType const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:1682:0
 #55 0x00007ffd34184469 clang::TypeVisitor<class clang::ASTNodeImporter, class llvm::Expected<class clang::QualType>>::Visit(class clang::Type const *) D:\test\build-lldb\tools\clang\include\clang\AST\TypeNodes.inc:49:0
 #56 0x00007ffd340b9a49 clang::ASTImporter::Import(class clang::Type const *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9166:0
 #57 0x00007ffd340b9c23 clang::ASTImporter::Import(class clang::QualType) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9180:0
 #58 0x00007ffd34156589 clang::ASTNodeImporter::import<class clang::QualType>(class clang::QualType const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:217:0
 #59 0x00007ffd34110b94 clang::ASTNodeImporter::import<class clang::TemplateArgument>(class clang::TemplateArgument const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:833:0
 #60 0x00007ffd341117f5 clang::ASTNodeImporter::import<class clang::TemplateArgumentLoc>(class clang::TemplateArgumentLoc const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:919:0
 #61 0x00007ffd34162f6b clang::ASTNodeImporter::importTemplateParameterDefaultArgument<class clang::TemplateTypeParmDecl>(class clang::TemplateTypeParmDecl const *, class clang::TemplateTypeParmDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:388:0
 #62 0x00007ffd340eee2d clang::ASTNodeImporter::VisitTemplateTypeParmDecl(class clang::TemplateTypeParmDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:5983:0
 #63 0x00007ffd341793b6 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:500:0
 #64 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
 #65 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
 #66 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
 #67 0x00007ffd341551ef clang::ASTNodeImporter::import<class clang::NamedDecl>(class clang::NamedDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #68 0x00007ffd341489b6 clang::ASTNodeImporter::ImportArrayChecked<class clang::NamedDecl *const *, class clang::NamedDecl **>(class clang::NamedDecl *const *, class clang::NamedDecl *const *, class clang::NamedDecl **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
 #69 0x00007ffd3414a439 clang::ASTNodeImporter::ImportContainerChecked<class clang::TemplateParameterList, class llvm::SmallVector<class clang::NamedDecl *, 4>>(class clang::TemplateParameterList const &, class llvm::SmallVector<class clang::NamedDecl *, 4> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
 #70 0x00007ffd34110500 clang::ASTNodeImporter::import<class clang::TemplateParameterList>(class clang::TemplateParameterList *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:798:0
 #71 0x00007ffd340f461c clang::ASTNodeImporter::VisitFunctionTemplateDecl(class clang::FunctionTemplateDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6734:0
 #72 0x00007ffd3417950e clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:536:0
 #73 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
 #74 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
 #75 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
 #76 0x00007ffd341551ef clang::ASTNodeImporter::import<class clang::NamedDecl>(class clang::NamedDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #77 0x00007ffd341090bc clang::ASTNodeImporter::VisitUnresolvedLookupExpr(class clang::UnresolvedLookupExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8630:0
 #78 0x00007ffd3417c2bc clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:180:0
 #79 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #80 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
 #81 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #82 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
 #83 0x00007ffd3410bc09 clang::ASTNodeImporter::VisitCallExpr(class clang::CallExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8713:0
 #84 0x00007ffd3417dbf3 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:614:0
 #85 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #86 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
 #87 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #88 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
 #89 0x00007ffd340f7006 clang::ASTNodeImporter::VisitIfStmt(class clang::IfStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7002:0
 #90 0x00007ffd341817cf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1604:0
 #91 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #92 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #93 0x00007ffd3415f0dc clang::ASTNodeImporter::importChecked<class clang::Stmt *>(class llvm::Error &, class clang::Stmt *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
 #94 0x00007ffd340f77de clang::ASTNodeImporter::VisitWhileStmt(class clang::WhileStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7057:0
 #95 0x00007ffd3417ba08 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:24:0
 #96 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
 #97 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
 #98 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
 #99 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#100 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#101 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#102 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#103 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#104 0x00007ffd340d8086 clang::ASTNodeImporter::ImportFunctionDeclBody(class clang::FunctionDecl *, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:3509:0
#105 0x00007ffd340e17c6 clang::ASTNodeImporter::VisitFunctionDecl(class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:4111:0
#106 0x00007ffd341787f5 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:256:0
#107 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#108 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#109 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#110 0x00007ffd34161b41 clang::ASTNodeImporter::importInto<class clang::FunctionDecl>(class clang::FunctionDecl *&, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:192:0
#111 0x00007ffd340f46c1 clang::ASTNodeImporter::VisitFunctionTemplateDecl(class clang::FunctionTemplateDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6739:0
#112 0x00007ffd3417950e clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:536:0
#113 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#114 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#115 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#116 0x00007ffd341551ef clang::ASTNodeImporter::import<class clang::NamedDecl>(class clang::NamedDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#117 0x00007ffd341090bc clang::ASTNodeImporter::VisitUnresolvedLookupExpr(class clang::UnresolvedLookupExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8630:0
#118 0x00007ffd3417c2bc clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:180:0
#119 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#120 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#121 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#122 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#123 0x00007ffd3410bc09 clang::ASTNodeImporter::VisitCallExpr(class clang::CallExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8713:0
#124 0x00007ffd3417dbf3 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:614:0
#125 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#126 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#127 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#128 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#129 0x00007ffd34100418 clang::ASTNodeImporter::VisitUnaryOperator(class clang::UnaryOperator *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7769:0
#130 0x00007ffd3418b7c9 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::VisitUnaryDeref(class clang::UnaryOperator *) D:\test\llvm-project\clang\include\clang\AST\StmtVisitor.h:164:0
#131 0x00007ffd3417b5f4 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\llvm-project\clang\include\clang\AST\StmtVisitor.h:94:0
#132 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#133 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#134 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#135 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#136 0x00007ffd340f86d6 clang::ASTNodeImporter::VisitReturnStmt(class clang::ReturnStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7147:0
#137 0x00007ffd3417edf5 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:944:0
#138 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#139 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#140 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
#141 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#142 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#143 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#144 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#145 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#146 0x00007ffd340d8086 clang::ASTNodeImporter::ImportFunctionDeclBody(class clang::FunctionDecl *, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:3509:0
#147 0x00007ffd340e17c6 clang::ASTNodeImporter::VisitFunctionDecl(class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:4111:0
#148 0x00007ffd341787f5 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:256:0
#149 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#150 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#151 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#152 0x00007ffd34161b41 clang::ASTNodeImporter::importInto<class clang::FunctionDecl>(class clang::FunctionDecl *&, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:192:0
#153 0x00007ffd340f46c1 clang::ASTNodeImporter::VisitFunctionTemplateDecl(class clang::FunctionTemplateDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6739:0
#154 0x00007ffd3417950e clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:536:0
#155 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#156 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#157 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#158 0x00007ffd341551ef clang::ASTNodeImporter::import<class clang::NamedDecl>(class clang::NamedDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#159 0x00007ffd341090bc clang::ASTNodeImporter::VisitUnresolvedLookupExpr(class clang::UnresolvedLookupExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8630:0
#160 0x00007ffd3417c2bc clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:180:0
#161 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#162 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#163 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#164 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#165 0x00007ffd3410bc09 clang::ASTNodeImporter::VisitCallExpr(class clang::CallExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8713:0
#166 0x00007ffd3417dbf3 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:614:0
#167 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#168 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#169 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#170 0x00007ffd340d43aa clang::ASTNodeImporter::ImportInitializer(class clang::VarDecl *, class clang::VarDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:2394:0
#171 0x00007ffd340e590f clang::ASTNodeImporter::VisitVarDecl(class clang::VarDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:4661:0
#172 0x00007ffd341789f9 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:296:0
#173 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#174 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#175 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#176 0x00007ffd3415435f clang::ASTNodeImporter::import<class clang::Decl>(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#177 0x00007ffd34111f61 clang::ASTNodeImporter::import<class clang::DeclGroupRef>(class clang::DeclGroupRef const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:964:0
#178 0x00007ffd34160c36 clang::ASTNodeImporter::importChecked<class clang::DeclGroupRef>(class llvm::Error &, class clang::DeclGroupRef const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#179 0x00007ffd340f5d06 clang::ASTNodeImporter::VisitDeclStmt(class clang::DeclStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6900:0
#180 0x00007ffd3418192b clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1628:0
#181 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#182 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#183 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
#184 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#185 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#186 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#187 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#188 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#189 0x00007ffd3415f0dc clang::ASTNodeImporter::importChecked<class clang::Stmt *>(class llvm::Error &, class clang::Stmt *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#190 0x00007ffd340f70a1 clang::ASTNodeImporter::VisitIfStmt(class clang::IfStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7005:0
#191 0x00007ffd341817cf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1604:0
#192 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#193 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#194 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
#195 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#196 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#197 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#198 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#199 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#200 0x00007ffd340d8086 clang::ASTNodeImporter::ImportFunctionDeclBody(class clang::FunctionDecl *, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:3509:0
#201 0x00007ffd340e17c6 clang::ASTNodeImporter::VisitFunctionDecl(class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:4111:0
#202 0x00007ffd341787f5 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:256:0
#203 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#204 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#205 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#206 0x00007ffd34161b41 clang::ASTNodeImporter::importInto<class clang::FunctionDecl>(class clang::FunctionDecl *&, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:192:0
#207 0x00007ffd340f46c1 clang::ASTNodeImporter::VisitFunctionTemplateDecl(class clang::FunctionTemplateDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6739:0
#208 0x00007ffd3417950e clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:536:0
#209 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#210 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#211 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#212 0x00007ffd341551ef clang::ASTNodeImporter::import<class clang::NamedDecl>(class clang::NamedDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#213 0x00007ffd341090bc clang::ASTNodeImporter::VisitUnresolvedLookupExpr(class clang::UnresolvedLookupExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8630:0
#214 0x00007ffd3417c2bc clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:180:0
#215 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#216 0x00007ffd340bc779 clang::ASTImporter::Import(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9720:0
#217 0x00007ffd3415478f clang::ASTNodeImporter::import<class clang::Expr>(class clang::Expr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#218 0x00007ffd3415e4fc clang::ASTNodeImporter::importChecked<class clang::Expr *>(class llvm::Error &, class clang::Expr *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#219 0x00007ffd3410bc09 clang::ASTNodeImporter::VisitCallExpr(class clang::CallExpr *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:8713:0
#220 0x00007ffd3417dbf3 clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:614:0
#221 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#222 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#223 0x00007ffd3415f0dc clang::ASTNodeImporter::importChecked<class clang::Stmt *>(class llvm::Error &, class clang::Stmt *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#224 0x00007ffd340f70a1 clang::ASTNodeImporter::VisitIfStmt(class clang::IfStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7005:0
#225 0x00007ffd341817cf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1604:0
#226 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#227 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#228 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
#229 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#230 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#231 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#232 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#233 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#234 0x00007ffd3415f0dc clang::ASTNodeImporter::importChecked<class clang::Stmt *>(class llvm::Error &, class clang::Stmt *const &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:693:0
#235 0x00007ffd340f70a1 clang::ASTNodeImporter::VisitIfStmt(class clang::IfStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:7005:0
#236 0x00007ffd341817cf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1604:0
#237 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#238 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#239 0x00007ffd34148416 clang::ASTNodeImporter::ImportArrayChecked<class clang::Stmt **, class clang::Stmt **>(class clang::Stmt **, class clang::Stmt **, class clang::Stmt **) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:705:0
#240 0x00007ffd3414a019 clang::ASTNodeImporter::ImportContainerChecked<class llvm::iterator_range<class clang::Stmt **>, class llvm::SmallVector<class clang::Stmt *, 8>>(class llvm::iterator_range<class clang::Stmt **> const &, class llvm::SmallVector<class clang::Stmt *, 8> &) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:719:0
#241 0x00007ffd340f6080 clang::ASTNodeImporter::VisitCompoundStmt(class clang::CompoundStmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6918:0
#242 0x00007ffd34181aaf clang::StmtVisitorBase<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Stmt *>>::Visit(class clang::Stmt *) D:\test\build-lldb\tools\clang\include\clang\AST\StmtNodes.inc:1652:0
#243 0x00007ffd340bc975 clang::ASTImporter::Import(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9738:0
#244 0x00007ffd3415684f clang::ASTNodeImporter::import<class clang::Stmt>(class clang::Stmt *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:204:0
#245 0x00007ffd340d8086 clang::ASTNodeImporter::ImportFunctionDeclBody(class clang::FunctionDecl *, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:3509:0
#246 0x00007ffd340e17c6 clang::ASTNodeImporter::VisitFunctionDecl(class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:4111:0
#247 0x00007ffd341787f5 clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:256:0
#248 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#249 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
#250 0x00007ffd340bb38f clang::ASTImporter::Import(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9530:0
#251 0x00007ffd34161b41 clang::ASTNodeImporter::importInto<class clang::FunctionDecl>(class clang::FunctionDecl *&, class clang::FunctionDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:192:0
#252 0x00007ffd340f46c1 clang::ASTNodeImporter::VisitFunctionTemplateDecl(class clang::FunctionTemplateDecl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:6739:0
#253 0x00007ffd3417950e clang::declvisitor::Base<struct std::add_pointer, class clang::ASTNodeImporter, class llvm::Expected<class clang::Decl *>>::Visit(class clang::Decl *) D:\test\build-lldb\tools\clang\include\clang\AST\DeclNodes.inc:536:0
#254 0x00007ffd340b93f7 clang::ASTImporter::ImportImpl(class clang::Decl *) D:\test\llvm-project\clang\lib\AST\ASTImporter.cpp:9134:0
#255 0x00007ffd2d593993 lldb_private::ClangASTImporter::ASTImporterDelegate::ImportImpl(class clang::Decl *) D:\test\llvm-project\lldb\source\Plugins\ExpressionParser\Clang\ClangASTImporter.cpp:1134:0
```

mizvekov commented:
Thanks, that narrows it down a lot!

I still wish I could test lldb locally. I will keep trying.

slydiman commented:
> I still wish I could test lldb locally. I will keep trying.

Please note lldb uses the system's `debugserver` on macOS. But on other systems (Linux, Windows) it uses its own lldb-server. You may not be able to reproduce the problem on macOS.

mizvekov commented:
@slydiman 

Thanks. You can see I made changes to our CI in this PR, and I managed to enable lldb, enable python testing and everything, but everything still passes in the lldb test suite.

This is a linux system, too it should be capable to reproduce this, right?

Can you help get this CI up to reproducing this problem?

mizvekov commented:
Nevermind, I managed to do it :)

mizvekov committed:
[clang] ASTContext: fix creation of UnaryTransformTypes

This makes sure UnaryTransformTypes are all uniqued,
which makes the AST node more well behaved, and helps
with aka suppression in diagnostics.

This also makes sure a canonical UnaryTransformType round trips
back to a canonical type when recreated from its components.

mizvekov committed:
[clang] ASTContext: make DecltypeType creation more well-behaved

This makes sure a canonical DecltypeType round-trips back to a
canonical type when rebuilt from its components.

mizvekov committed:
fix structural comparison of DependentTemplateStorage
